### PR TITLE
Feature/tech 901: oauth middleware and backend session

### PR DESCRIPTION
**Pull request by lruzicki**
Ticket:
https://govstack-global.atlassian.net/browse/TECH-901

Changes:
- middleware for auth is added
- oauth uses backend session

Tests:
- manually tested the middleware and it worked as intended

### CODE REVIEW

File: backend/src/config/index.ts [hunk 1]
```diff
@@ -62,6 +64,8 @@ const appConfig: AppConfig = {
     clientSecret: process.env.GITHUB_CLIENT_SECRET!,
     callbackUrl: process.env.GITHUB_CALLBACK_URL!,
     devLoginMode: process.env.GITHUB_DEV_LOGIN_MODE ? process.env.GITHUB_DEV_LOGIN_MODE === 'true' : false,
+    jwtSecret: process.env.GITHUB_JWT_SECRET!,
+    tokenExpirationTime: 1 * 2 * 60 * 60, // 2 hours in seconds
```
Comment (dborowiecki) [inline backend/src/config/index.ts:68]: change it to 8 hours

File: backend/src/db/pipelines/compliance/softwareDetailAggregation.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export const softwareDetailAggregationPipeline = (softwareName: string): any[] =
             "bbVersion": "$$detail.v.bbVersion",
             "requirements": "$$detail.v.requirementSpecificationCompliance",
             "interface": "$$detail.v.interfaceCompliance",
-            "deploymentCompliance": "$deploymentCompliance.requirements" 
+            "deploymentCompliance": "$$detail.v.deploymentCompliance"
```
Comment (dborowiecki) [inline backend/src/db/pipelines/compliance/softwareDetailAggregation.ts:19]: is second dolar sign intentional?

**Review by dborowiecki — CHANGES_REQUESTED** [review]

File: backend/src/db/pipelines/compliance/softwareDetailAggregation.ts (hunk 2 above)
Comment (lruzicki) [reply backend/src/db/pipelines/compliance/softwareDetailAggregation.ts:19]: No, it is a typo. It work in both with single and double but the single one is a proper syntax

File: backend/src/config/index.ts (hunk 1 above)
Comment (lruzicki) [reply backend/src/config/index.ts]: okay, done

**Review by dborowiecki — APPROVED** [review]

**Pull request merged**